### PR TITLE
app: location: Use zbus subscriber instead of listener

### PR DESCRIPTION
**Pull request by gregersrygg**
Updated the location module to use zbus subscribe instead of a listener for consistency and easier logic. Removed semaphores and modem lib connection handling. Connection handling might be needed unless the location lib doesn't handle this internally, but it should probably be dealt with using the Zephyr State Machine Framework.

### CODE REVIEW

File: app/src/modules/location/location.c [hunk 1]
```diff
@@ -59,74 +72,49 @@ void location_task(void)
 	}
 
 	err = location_init(location_event_handler);
-	if (err)
-	{
+	if (err) {
 		LOG_ERR("Unable to init location library: %d", err);
```
Comment (simensrostad) [inline app/src/modules/location/location.c:76]:
```suggestion
		LOG_ERR("Unable to init location library: %d", err);
		SEND_FATAL_ERROR();
```

File: app/src/modules/location/location.c [hunk 2]
```diff
@@ -22,13 +22,10 @@
...
 BUILD_ASSERT(CONFIG_APP_LOCATION_WATCHDOG_TIMEOUT_SECONDS >
-			 CONFIG_APP_LOCATION_TRIGGER_TIMEOUT_SECONDS,
+			 CONFIG_APP_LOCATION_ZBUS_TIMEOUT_SECONDS,
 			 "Watchdog timeout must be greater than trigger timeout");
 
-
-static K_SEM_DEFINE(location_lib_sem, 1, 1);
-static K_SEM_DEFINE(modem_init_sem, 0, 1);
-static K_SEM_DEFINE(trigger_sem, 0, 1);
+ZBUS_SUBSCRIBER_DEFINE(location, CONFIG_APP_LOCATION_ZBUS_QUEUE_SIZE);
```
Comment (simensrostad) [inline app/src/modules/location/location.c:29]:
```suggestion

/* Forward declarations */
```

File: app/src/modules/location/location.c [hunk 3]
```diff
@@ -22,13 +22,10 @@
 LOG_MODULE_REGISTER(location_module, CONFIG_APP_LOCATION_LOG_LEVEL);
 
 BUILD_ASSERT(CONFIG_APP_LOCATION_WATCHDOG_TIMEOUT_SECONDS >
-			 CONFIG_APP_LOCATION_TRIGGER_TIMEOUT_SECONDS,
```
Comment (simensrostad) [inline app/src/modules/location/location.c:25]: Nit: Some alignment issues here it looks like. Could align with CONFIG_APP_LOCATION

File: app/src/modules/location/location.c [hunk 4]
```diff
@@ -59,74 +72,49 @@ void location_task(void)
...
 		}
 
-		location_config_defaults_set(&config, 0, NULL);
-		config.mode = LOCATION_REQ_MODE_ALL;
-		err = location_request(&config);
-		if (err)
-		{
-			LOG_ERR("Unable to send location request: %d", err);
+		if (&TRIGGER_CHAN == chan) {
+			LOG_WRN("Trigger received");
```
Comment (simensrostad) [inline app/src/modules/location/location.c:108]:
```suggestion
			LOG_DBG("Trigger received");
```

**Review by simensrostad — COMMENTED** [review]
Looks good,

I get an unknown event when testing, was fixed by adding a handler case for:

	case **LOCATION_EVT_STARTED**:
		LOG_WRN("Getting location started");
		break;
	default:
		LOG_DBG("Unknown event %d", event_data->id);
		break;

File: app/src/modules/location/location.c (hunk 2 above)
Comment (gregersrygg) [reply app/src/modules/location/location.c:29]: This is unrelated to the PR.

**Pull request merged**